### DISCUSSION TITLE
Using supabase Storage to store images

### DISCUSSION THREAD

**Discussion by joshuabourke** [supabase/supabase#12844] [Questions]
Hi All, 

I want to use a Supabase database using swift and Swiftui.

I am trying to CRUD with images and some data plus auth users. Im pretty much just trying to set up a whole backend 😅 for my application.

I seem to be getting a lot of errors back from the functions I am writing. They must be coming back from Supabase they read "StorageError(statusCode: Optional(400), message: Optional("Error"))"

Any help will be much appreciated.

Thanks Josh

**Comment by GaryAustin1** [comment]
You'll need to try and narrow down the supabase call getting the error.  It is likely a call to supabase storage but too generic without more info.   Do you have RLS policies setup on storage.objects or buckets depending on what you are doing?  Although that probably is a different code.

**Reply by joshuabourke** [reply]
I am getting error on the path.

I am trying to path into a folder in my storage bucket. the path "userId/" I thought was meant to collect all the files in the storage bucket and then in the folder of the users id that is where I am getting the error. Also when listing using that same path I get nothing. It works when i path straight to the file i am wanting to get.

**Reply by GaryAustin1** [reply]
There are no real folders in storage, it is just a processing of the pathname using / as a seperator.  Is it a public bucket?  If so that is how you might get the file directly.  To use list you have to have a select policy that you meet.  You can use a string in that to represent a path to the files.

You should show your RLS policies and be specific on private/public and show some sample pathnames if that does not help.

**Reply by joshuabourke** [reply]
<img width="1169" alt="screenshotpolicy" src="https://user-images.githubusercontent.com/24494726/223282403-17e17cf5-ae7f-4d27-92eb-2d1b4f526e6a.png">

I have attached an image of the policies I am using. It works when setting the exact path from where to download an image from. The problem is it isn't listing all the images for me to be able to download them all. I am wanting to store user images on Supabase and fetch them to display them in the app. Everything else works just cant get my head around this list.

**Reply by GaryAustin1** [reply]
The main policy here for list to work is the select policy on that bucket and path.  You don't show your policies just labels and no example file path.
You did not say if public or private (if private and you could read the file, then likely means you should be able to list)
You should show your list command.
